src/harness/util.py

```diff
@@ -285,3 +292,98 @@ def convertRequestToRequestWithCpiSignature(private_key, cpi_id,
   request['cpiSignatureData']['protectedHeader'] = jwt_message[0]
   request['cpiSignatureData']['encodedCpiSignedData'] = jwt_message[1]
   request['cpiSignatureData']['digitalSignature'] = jwt_message[2]
+
+class QueryPropagationAntennaModel:
+    nlcd_driver = nlcd.NlcdDriver()
+       
+    def getPPinPPA(self, boundarysPoints, ARCSEC):   
+        poly = geometry.Polygon([[lon,lat] for lon,lat in boundarysPoints])
+        arcsec = ARCSEC/3600.0
+        minlon, minlat, maxlon, maxlat = poly.bounds
+        
+        iLat = np.floor(minlat)
+        iLon = np.floor(minlon)
+        nL = np.floor((minlat - iLat)/arcsec)
+        nH = np.ceil((maxlat - iLat)/arcsec)
+        mL = np.floor((minlon - iLon)/arcsec)
+        mH = np.ceil((maxlon - iLon)/arcsec)
+            
+        latgrid = np.arange(iLat + nL*arcsec, iLat + nH*arcsec, arcsec)
+        longrid = np.arange(iLon + mL*arcsec, iLon + mH*arcsec, arcsec)
+        
+        if len(latgrid) > 0 and len(latgrid)>0:
+            ppa = []
+    
+            for lat in latgrid:
+                for lon in longrid:
+                    ppa_check_point = Point(lon, lat)
+                    if ppa_check_point.within(poly):
+                            ppa.extend([[lat, lon]])
+    
+            if (len(ppa) == 1):
+                return 0, ppa[0][0], ppa[0][1]
+            else:
+                return -1, -1, -1
+        else:
+            return -1, -1, -1
+    
+    def computePropagationAntennaModel(self, requestJson):
```

Comment:
As said earlier, this function should be a static function, not a method.

The name needs to be updated: ComputePropagationAntennaModel()